HTML parser: only adjust approved SVG and MathML attributes

annevk opened the pull request:
#### 83e92a3d7ab3fcddacde92972ea1aade639459d3
<pre>
HTML parser: only adjust approved SVG and MathML attributes
<a href="https://bugs.webkit.org/show_bug.cgi?id=254755">https://bugs.webkit.org/show_bug.cgi?id=254755</a>
rdar://107428878

Reviewed by Chris Dumez.

Instead of the HTML parser using global lists of SVG, MathML, XLink, and XML attributes, define them locally as the HTML Standard does. As the HTML parser is the sole consumer of these lists that allows them to be smaller and also makes it clearer that any new attributes should follow HTML naming conventions.

Additionally this removes a number of unused attributes:

- SVG contentScriptType
- SVG contentStyleType
- SVG externalResourcesRequired
- XML xml:base

* LayoutTests/html5lib/resources/tests11.dat:
* LayoutTests/html5lib/resources/webkit02.dat:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=uri-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=write-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=write_single-expected.txt:
* Source/WebCore/WebCore.order:
* Source/WebCore/dom/make_names.pl:
(printConstructorInterior):
(upperCaseName):
(printNamesHeaderFile):
(printNamesCppFile):
(printWrapperFactoryHeaderFile):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::createSVGAttributesMap):
(WebCore::adjustSVGAttributes):
(WebCore::createMathMLAttributesMap):
(WebCore::adjustMathMLAttributes):
(WebCore::createForeignAttributesMap):
(WebCore::createCaseMap): Deleted.
* Source/WebCore/mathml/mathattrs.in:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::addEventListener):
(WebCore::SVGElement::childShouldCreateRenderer const):
(WebCore::SVGElement::createAnimator):
(WebCore::SVGElement::animatableAttributeForName):
* Source/WebCore/svg/svgattrs.in:
* Source/WebCore/xml/xmlattrs.in:

Canonical link: <a href="https://commits.webkit.org/262502@main">https://commits.webkit.org/262502@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3fc4f3b4cce92013b13ca32e405f17e92a63ad1f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1662 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1693 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1751 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2585 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1785 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1643 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1757 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1562 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1677 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1508 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1497 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2422 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1500 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1483 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1404 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1522 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1525 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2589 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1555 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1385 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1473 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1490 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/418 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1619 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/WebCore.order:
```diff
@@ -9896,13 +9896,11 @@ __ZN7WebCore9nullQNameEv
 __ZN3WTF13hashTableSwapINS_12AtomicStringEN7WebCore13QualifiedNameEEEvRNS_12KeyValuePairIT_T0_EES8_
 __ZNK3WTF7HashMapINS_12AtomicStringEN7WebCore13QualifiedNameENS_16AtomicStringHashENS_10HashTraitsIS1_EENS5_IS3_EEE3getERKS1_
 __ZN7WebCoreL23adjustForeignAttributesEPNS_15AtomicHTMLTokenE
-__ZN7WebCore10XLinkNames13getXLinkAttrsEv
```
Note for the future, you don't need to update order files.

annevk replied:
Thanks, I probably misdiagnosed a bot failure on an earlier PR.

cdumez commented on Source/WebCore/svg/SVGElement.cpp:
```diff
@@ -606,7 +606,7 @@ bool SVGElement::childShouldCreateRenderer(const Node& child) const
     case ElementNames::SVG::tspan:
         return false;
     default:
-        break;        
```
Please update your IDE to not do this. This is generally frown upon in the WebKit project, though I won't ask you to revert from this patch.

annevk replied:
Discussions with others led me to believe this was okay. I tagged you on the relevant Slack thread.

cdumez reviewed (approved):
LGTM

annevk committed:
HTML parser: only adjust approved SVG and MathML attributes
https://bugs.webkit.org/show_bug.cgi?id=254755
rdar://107428878

Reviewed by Chris Dumez.

Instead of the HTML parser using global lists of SVG, MathML, XLink, and XML attributes, define them locally as the HTML Standard does. As the HTML parser is the sole consumer of these lists that allows them to be smaller and also makes it clearer that any new attributes should follow HTML naming conventions.

Additionally this removes a number of unused attributes:

- SVG contentScriptType
- SVG contentStyleType
- SVG externalResourcesRequired
- XML xml:base

* LayoutTests/html5lib/resources/tests11.dat:
* LayoutTests/html5lib/resources/webkit02.dat:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=uri-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=write-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11_run_type=write_single-expected.txt:
* Source/WebCore/WebCore.order:
* Source/WebCore/dom/make_names.pl:
(printConstructorInterior):
(upperCaseName):
(printNamesHeaderFile):
(printNamesCppFile):
(printWrapperFactoryHeaderFile):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::createSVGAttributesMap):
(WebCore::adjustSVGAttributes):
(WebCore::createMathMLAttributesMap):
(WebCore::adjustMathMLAttributes):
(WebCore::createForeignAttributesMap):
(WebCore::createCaseMap): Deleted.
* Source/WebCore/mathml/mathattrs.in:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::addEventListener):
(WebCore::SVGElement::childShouldCreateRenderer const):
(WebCore::SVGElement::createAnimator):
(WebCore::SVGElement::animatableAttributeForName):
* Source/WebCore/svg/svgattrs.in:
* Source/WebCore/xml/xmlattrs.in:

Canonical link: https://commits.webkit.org/262502@main
